phoenix-pherf/src/main/java/org/apache/phoenix/pherf/workload/mt/tenantoperation/TenantOperationFactory.java

```diff
@@ -0,0 +1,501 @@
+import org.apache.phoenix.pherf.configuration.Upsert;
+import org.apache.phoenix.pherf.configuration.UserDefined;
+import org.apache.phoenix.pherf.configuration.XMLConfigParser;
+import org.apache.phoenix.pherf.rules.DataValue;
+import org.apache.phoenix.pherf.rules.RulesApplier;
+import org.apache.phoenix.pherf.util.PhoenixUtil;
+import org.apache.phoenix.pherf.workload.mt.EventGenerator;
+import org.apache.phoenix.pherf.workload.mt.NoopOperation;
+import org.apache.phoenix.pherf.workload.mt.Operation;
+import org.apache.phoenix.pherf.workload.mt.OperationStats;
+import org.apache.phoenix.pherf.workload.mt.PreScenarioOperation;
+import org.apache.phoenix.pherf.workload.mt.QueryOperation;
+import org.apache.phoenix.pherf.workload.mt.UpsertOperation;
+import org.apache.phoenix.pherf.workload.mt.UserDefinedOperation;
+import org.apache.phoenix.util.EnvironmentEdgeManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.math.BigDecimal;
+import java.sql.Array;
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Factory class for operations.
+ * The class is responsible for creating new instances of various operation types.
+ * Operations typically implement @see {@link TenantOperationImpl}
+ * Operations that need to be executed are generated
+ * by @see {@link EventGenerator}
+ */
+public class TenantOperationFactory {
+
+    private static class TenantView {
+        private final String tenantId;
+        private final String viewName;
+
+        public TenantView(String tenantId, String viewName) {
+            this.tenantId = tenantId;
+            this.viewName = viewName;
+        }
+
+        public String getTenantId() {
+            return tenantId;
+        }
+
+        public String getViewName() {
+            return viewName;
+        }
+    }
+    private static final Logger LOGGER = LoggerFactory.getLogger(TenantOperationFactory.class);
+    private final PhoenixUtil phoenixUtil;
+    private final DataModel model;
+    private final Scenario scenario;
+    private final XMLConfigParser parser;
+
+    private final RulesApplier rulesApplier;
+    private final LoadProfile loadProfile;
+    private final List<Operation> operationList = Lists.newArrayList();
+
+    private final BloomFilter<TenantView> tenantsLoaded;
+
+    public TenantOperationFactory(PhoenixUtil phoenixUtil, DataModel model, Scenario scenario) {
+        this.phoenixUtil = phoenixUtil;
+        this.model = model;
+        this.scenario = scenario;
+        this.parser = null;
+        this.rulesApplier = new RulesApplier(model);
+        this.loadProfile = this.scenario.getLoadProfile();
+        Funnel<TenantView> tenantViewFunnel = new Funnel<TenantView>() {
```

Comment:
Can we break this method up too instead of putting it all in the constructor itself?